src/job/mod.rs

```diff
@@ -53,20 +53,21 @@ pub async fn init() -> Result<(), anyhow::Error> {
         .await;
     }
 
+    let res = db::get_instance().await;
+    let instance_id;
+    if res.as_ref().is_err() || res.as_ref().unwrap().is_none() {
+        instance_id = ider::generate();
+        let _ = db::set_instance(&instance_id).await;
+    } else {
+        instance_id = res.unwrap().unwrap();
+    }
```

Comment:
1.
```suggestion
    let instance_id = match db::get_instance().await {
        Ok(Some(instance)) => instance,
        Ok(None) | Err(_) => {
            let id = ider::generate();
            let _ = db::set_instance(&id).await;
            id
        }
    };
```

2. `db::get_instance` never returns `Ok(None)`. Can we change its return type to `Result<String, anyhow::Error>`?

https://github.com/zinclabs/zincobserve/blob/18152e4d16eb6cfccd03d9ab8f0f1b5c9749cbed/src/service/db/mod.rs#L95-L104

Reply:
will take this up in next PR